python/pyiceberg/avro/reader.py

```diff
@@ -104,16 +104,22 @@ def _skip_map_array(decoder: BinaryDecoder, skip_entry: Callable[[], None]) -> N
         block_count = decoder.read_int()
 
 
-@dataclass(frozen=True)
 class AvroStruct(StructProtocol):
-    _data: List[Union[Any, StructProtocol]] = dataclassfield()
+    _data: List[Union[Any, StructProtocol]]
```

Comment:
Is anything here specific to Avro? Or should this be a generic struct in a different place?